Removing the check of the officer_note property.

Pyrodius0 opened the pull request:
fixes #29 
resolves #29 
closes #29

Lemmings19 commented:
_"This is a pretty quick fix."_

**Bwahahahaha!**

Need to fix clearing the officer note when updating a member:
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L177
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L204

When just updating their notes:
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L264
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L284

When creating a character:
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/CharacterController.php#L91

Updating a character:
https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/CharacterController.php#L314

I was worried about some edge cases where it would try to populate based on `request()->input('officer_note')`, but that field wouldn't actually be on the page if a user wasn't an officer or GM. However after testing it looks like you can call `var = request()->input('some_fake_field')` and it allows it, so that'll never happen.

Pyrodius0 commented:
> _"This is a pretty quick fix."_
> 
> **Bwahahahaha!**
> 
> Need to fix clearing the officer note when updating a member:
> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L177
Updated this one. 


> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L204
I do not believe this one needs to be updated.. as it is just comparing if the updated value is not equal to the current officer note. That will be true even if its null or not null. 

> 
> When just updating their notes:
> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L264
updated

> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L284
I do not believe this one needs to be updated.. as it is just comparing if the updated value is not equal to the current officer note. That will be true even if its null or not null. 

> 
> When creating a character:
> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/CharacterController.php#L91
Updated

> 
> Updating a character:
> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/CharacterController.php#L314
Updated

Lemmings19 commented:
>> https://github.com/thatsmybis/thatsmybis/blob/master/app/Http/Controllers/MemberController.php#L204
> I do not believe this one needs to be updated.. as it is just comparing if the updated value is not equal to the current officer note. That will be true even if its null or not null.

My mistake. I was thinking it was checking against `request()->input('officer_note')` and not `$updateValues['officer_note']`. The former would allow someone to add the `officer_note` field to the page, submit it, and have an audit log saying they updated/changed the officer note.
